scopes/scopes_test.go

```diff
@@ -0,0 +1,113 @@
+package scopes
+
+import (
+	"testing"
+)
+
+func accept(t *testing.T, givenScopes Given, requiredScopes Required) {
+	if !givenScopes.Satisfies(&requiredScopes) {
+		t.Errorf("Expected given scopes %q to satisfy required scopes %q, but did not.", givenScopes, requiredScopes)
+	}
+}
+
+func reject(t *testing.T, givenScopes Given, requiredScopes Required) {
+	if givenScopes.Satisfies(&requiredScopes) {
+		t.Errorf("Expected given scopes %q *not* to satisfy required scopes %q, but it did.", givenScopes, requiredScopes)
+	}
+}
+
+func TestScopes1(t *testing.T) {
+	accept(
+		t,
+		[]string{
+			"abc:*",
+			"123:4:56",
+			"xyz",
+			"AB:*",
+		},
+		[][]string{
+			{"abc:def", "AB:CD:EF"},
+			{"123:4:5"},
+			{"abc:def", "123:4"},
+			{"Xxyz"},
+		},
+	)
+}
```

Comment:
This is successfully testing the disjunction: `{"abc:def", "AB:CD:EF"}` is satisfied, but the remainder aren't, yet the result is true.  I don't see anything testing the conjunction, though: `{{"a", "b"}}` is satisfied by `{"a", "b"}` but NOT by `{"a"}` or `{"b"}` or `{}`.  Maybe I'm missing that -- it would help to have comments or longer test names to suggest what each one is looking for.